packages/app/src/app/hooks/useScript.ts

```diff
@@ -0,0 +1,60 @@
+// Based on https://usehooks.com/useScript/
+import { useState, useEffect } from 'react';
+
+const cachedScripts = [];
+
+export const useScript = (src: string) => {
+  const [state, setState] = useState({
+    loaded: false,
+    error: false,
+  });
+
+  useEffect(() => {
+    if (cachedScripts.includes(src)) {
+      setState({
+        loaded: true,
+        error: false,
+      });
+    } else {
+      cachedScripts.push(src);
+
+      const script = document.createElement('script');
+      script.src = src;
+      script.async = true;
+
+      const onScriptLoad = () => {
+        setState({
+          loaded: true,
+          error: false,
+        });
+      };
+
+      const onScriptError = () => {
+        const index = cachedScripts.indexOf(src);
+
+        if (index >= 0) cachedScripts.splice(index, 1);
+
+        script.remove();
+
+        setState({
+          loaded: true,
+          error: true,
+        });
+      };
+
+      script.addEventListener('load', onScriptLoad);
+      script.addEventListener('error', onScriptError);
+
+      document.body.appendChild(script);
+
+      return () => {
+        script.removeEventListener('load', onScriptLoad);
+        script.removeEventListener('error', onScriptError);
+      };
```

Comment:
Shouldn't the script be detached from `document.body` too? 🤔

Reply:
Hmm, maybe, I'm not entire sure what the possible side-effects would be of removing a script from a page. My intuition tells me that it could cause unwanted behavior.

Reply:
Not much would happen if you remove it, on the other hand, I don't see a reason to remove it.